client/app/telecom/pack/xdsl/access/comfortExchange/pack-xdsl-access-comfort-exchange.controller.js

```diff
@@ -0,0 +1,82 @@
+import { COMFORT_EXCHANGE_TYPE_ERROR } from './pack-xdsl-access-comfort-exchange.constant';
+
+export default class XdslAccessComfortExchangeCtrl {
+  /* @ngInject */
+  constructor($q, $translate, OvhApiXdsl, TucToast, TucToastError) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.OvhApiXdsl = OvhApiXdsl;
+    this.TucToast = TucToast;
+    this.TucToastError = TucToastError;
+  }
+
+  $onInit() {
+    this.exchange = {
+      order: {
+        url: null,
+        id: null,
+        priceWithoutTax: null,
+        priceWithTax: null,
+        tax: null,
+      },
+      isSuccess: false,
+    };
+    this.isAvailable = false;
+    this.getListOpenedRMA();
+  }
+
+  getListOpenedRMA() {
+    this.isRMAOpened = false;
+    return this.OvhApiXdsl.RMA()
+      .v6()
+      .query({
+        xdslId: this.xdslId,
+      })
+      .$promise.then((rmas) => {
+        if (rmas.length > 0) {
+          this.rmas = [];
+          this.isRMAOpened = true;
+          rmas.forEach(id => this.OvhApiXdsl.RMA()
```

Comment:
You should use Iceberg for this
 ```suggestion
          this
            .OvhApiXdsl
            .Iceberg()
            .query()
            .expand('CachedObjectList-Pages')
            .execute()
            .$promise
```
Icerberg will do the same but in the API directly, so the result will be way way faster !

First you have to add Icerberg support for your api call, like this: https://github.com/ovh-ux/ovh-api-services/blob/master/src/api/support/support.iceberg.service.js